### PR TITLE
testers.hasPkgConfigModules: misc fixes

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 284652` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>147 packages built:</summary>
  <ul>
    <li>tests.pkg-config.defaultPkgConfigPackages.IL</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ImageMagick</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.MagickWand</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Concurrent</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Core</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5DBus</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Gui</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Network</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5OpenGL</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5OpenGLExtensions</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5PrintSupport</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Qml</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5QmlModels</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Quick</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5QuickTest</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5QuickWidgets</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Sql</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Test</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Widgets</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.Qt5Xml</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.SoapySDR</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.alsa</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.alsa-topology</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.bzip2</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.cairo-gobject</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.cairo-pdf</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.cairo-ps</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.cairo-svg</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.egl</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.fftw3</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.fftw3f</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.form</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.formw</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.freealut</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.freetype2</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.geos</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.gl</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.glew</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.glu</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.glut</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.gnome-keyring-1</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.hidapi-hidraw</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.hidapi-libusb</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.icu-i18n</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.icu-io</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.icu-uc</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.imlib2</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.jack</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.lapack</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libR</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libavutil</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libb2</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libbrotlidec</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libbrotlienc</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libcrypto</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libcurl</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libecpg</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libecpg_compat</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libgsasl</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libidn</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libjpeg</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.liblzma</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libmagic</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libmnl</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libnotify</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpcap</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpcre</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpcre2-16</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpcre2-32</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpcre2-8</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpcre2-posix</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpgtypes</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpng</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpq</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpulse</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpulse-mainloop-glib</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libpulse-simple</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libqrencode</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.librtlsdr</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libsass</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libsctp</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libsecp256k1</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libsodium</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libssh2</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libssl</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libstatgrab</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libsystemd</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libturbojpeg</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libudev</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libzip</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libzmq</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.menu</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.menuw</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ncurses</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ncurses++</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ncurses++w</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ncursesw</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.netsnmp</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.nix-cmd</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.nix-expr</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.nix-main</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.nix-store</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.odbc</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-c</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-cxx</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-f77</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-f90</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-fort</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.openblas</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.openssl (tests.testers.hasPkgConfigModules.openssl-has-openssl)</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.orte</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.panel</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.panelw</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.pangocairo</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.poppler-glib</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.python</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.sdl2</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.sndfile</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.sqlite3</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.taglib</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.taglib_c</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.tdjson</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.tensorflow</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.uuid</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.wayland-client</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.wayland-cursor</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.wayland-scanner</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.wayland-server</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.x11</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xau</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xcursor</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xerces-c</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xext</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xft</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xi</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xinerama</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xkbcommon</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xpm</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xrandr</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xrender</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xscrnsaver</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xtst</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.xxf86vm</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.zlib (tests.testers.hasPkgConfigModules.zlib-has-meta-pkgConfigModules ,tests.testers.hasPkgConfigModules.zlib-has-zlib)</li>
    <li>tests.testers.hasPkgConfigModules.openssl-has-all-meta-pkgConfigModules</li>
    <li>tests.testers.hasPkgConfigModules.zlib-does-not-have-ylib</li>
  </ul>
</details>

File: pkgs/build-support/testers/hasPkgConfigModules/tester.nix [hunk 1]
```diff
@@ -30,18 +30,19 @@ runCommand testName {
         }
         package.meta;
   } ''
+    touch "$out"
```
Comment (sternenseemann) [inline pkgs/build-support/testers/hasPkgConfigModules/tester.nix:33]: I don't see an use case for hasPkgConfigModules with `moduleNames = [ ];`.

Comment (NickCao) [reply pkgs/build-support/testers/hasPkgConfigModules/tester.nix:33]: Some generated codes do use this, namely xorg: https://github.com/NixOS/nixpkgs/pull/280122

Comment (NickCao) [reply pkgs/build-support/testers/hasPkgConfigModules/tester.nix:33]: Got in as part of https://github.com/NixOS/nixpkgs/pull/284740

Comment (sternenseemann) [reply pkgs/build-support/testers/hasPkgConfigModules/tester.nix]: I see.

What about

```nix
if moduleNames == [ ]
then lib.warn "hasPkgConfigModules: No modules specified for ${package.name}" pkgs.emptyFile # (or `null`?)
else runCommandNoCC …
```

The passthru stuff shouldn't show up in ofborg and I think it's a bit clearer and avoids useless derivations.

Comment (NickCao) [reply pkgs/build-support/testers/hasPkgConfigModules/tester.nix]: Do we really need the warnings?

Comment (sternenseemann) [reply pkgs/build-support/testers/hasPkgConfigModules/tester.nix]: My reasoning is that having this tester enabled implies that there should be some modules *usually*. What do you think @roberth?

**Pull request merged**